### PR TITLE
THP preparation

### PR DESCRIPTION
**Pull request by M1nd3r**
Minor changes from the THP branch that can be merged already.

- docs: fixes in documentation and some typo corrections
- protob: reservation of wire types for THP messages in the `messages.proto` file
- build: changes in Makefile and SConscripts affecting THP only
- protob: allow encoding of protobuf messages into memoryview
- refactor: use TYPE CHECKING instead of direct import and fix style of type ignore statement
- codec: remove unused session_id (it is/was always a constant 0)

### CODE REVIEW

File: core/src/trezor/ui/layouts/common.py [hunk 1]
```diff
@@ -64,7 +64,7 @@ def raise_if_not_confirmed(
     exc: ExceptionType = ActionCancelled,
 ) -> Awaitable[None]:
     action = interact(layout_obj, br_name, br_code, exc)
-    return action  # type: ignore ["UiResult" is incompatible with "None"]
+    return action  # type: ignore [Expression of type "Coroutine[Any, Any, UiResult]" is incompatible with return type "Awaitable[None]"]
```
Comment (mmilata) [inline core/src/trezor/ui/layouts/common.py:67]: Was pyright unhappy about this, and why didn't CI catch it?

**Review by mmilata — APPROVED** [review]
LGTM, nothing controversial here

File: core/src/trezor/ui/layouts/common.py (hunk 1 above)
Comment (M1nd3r) [reply core/src/trezor/ui/layouts/common.py:67]: I was having some problems locally, but CI is (probably) happy with both versions. Do you want to have the old version back  (i.e. do not fix it if it is not broken) or is it ok as I have it in this PR?

File: common/protob/messages.proto [hunk 2]
```diff
@@ -324,4 +324,7 @@ enum MessageType {
     MessageType_BenchmarkNames = 9101 [(bitcoin_only) = true];
     MessageType_BenchmarkRun = 9102 [(bitcoin_only) = true];
     MessageType_BenchmarkResult = 9103 [(bitcoin_only) = true];
+
+    // THP
+    reserved 1000 to 1099; // See messages-thp.proto
```
[the remark this answers is not in the document]
Comment (M1nd3r) [reply common/protob/messages.proto]: fixed here: https://github.com/trezor/trezor-firmware/commit/a9d81b355bab5e7788ca993738473fe43277215b

File: core/src/trezor/ui/layouts/common.py (hunk 1 above)
Comment (M1nd3r) [reply core/src/trezor/ui/layouts/common.py]: Returned back to the previous state (as in `main`)

**Pull request merged**